cliccl: use latest version for the TestList

RaduBerinde opened the pull request:
We were using `MakeClusterSettings()` which initializes the cluster at
the minimum supported version. Switching to
`MakeTestingClusterSettings()` which initializes at the latest
version. It makes more sense for the test to change when the latest
version changes.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/136820)
<!-- Reviewable:end -->

itsbilal reviewed (approved):
LGTM

RaduBerinde committed:
cliccl: use latest version for the TestList

We were using `MakeClusterSettings()` which initializes the cluster at
the minimum supported version. Switching to
`MakeTestingClusterSettings()` which initializes at the latest
version. It makes more sense for the test to change when the latest
version changes.

Epic: none
Release note: None

RaduBerinde commented:
TFTR!

bors r+
